### ISSUE TITLE
Disallow labels on sequence flows that are not outgoing from OR/XOR gateways

### ISSUE THREAD

**Issue by till-stadtler** [bpmn-io/bpmnlint#123]
#### The rule should detect the following modeling patterns

Sequence flows that are not outgoing from OR/XOR gateways should not have labels. Labeling these sequence flows can lead to confusion.
![no-labels-on-standard-sequence-flows-1](https://github.com/bpmn-io/bpmnlint/assets/117343024/27558465-aa2d-4537-a13f-27e658cd41bb)

Instead either use no label, because the following activity speaks for itself, or an intermediate none event:
![no-labels-on-standard-sequence-flows-2](https://github.com/bpmn-io/bpmnlint/assets/117343024/21430c5d-1c12-4006-b8b5-2d069159e825)

#### How does the rule improve the BPMN diagram?

Instead of having confusing labels on sequence flows that are not outgoing from OR/XOR gateways, they are either removed or replaced with an intermediate none event.

In Camunda, intermediate none events can be used for business intelligence in Optimize.

#### Rule Details

* Name: no-isolated-sequence-flow-label
* Default notification level: `warn`

#### What alternatives did you consider? 

BPMN models continue to contain labels on sequence flows that are not outgoing from OR/XOR gateways.

**Comment by nikku** [comment]
I assume this does not apply to conditional labels, right?

![image](https://github.com/bpmn-io/bpmnlint/assets/58601/02521346-efd7-496e-af16-b19b07748816)

**Comment by till-stadtler** [comment]
You are right! Labels should also be allowed on conditional sequence flows!

**Comment by nikku** [comment]
Will be addressed via https://github.com/bpmn-io/bpmnlint/pull/129 (some more work needed to wrap that one up, some day maybe).
